### PR TITLE
feat: create a child module for all DA patterns

### PR DESCRIPTION
**Pull request by akocbek**
### Description

https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/issues/402

#### Changes that affect the core Terraform module or submodules
- [ ] Bug fix
- [x] New feature
- [ ] Dependency update

#### Changes that don't affect the core Terraform module or submodules
- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other

#### Release required?
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

File: patterns/vpc/module/variables.tf
Comment (toddgiguere) [reply patterns/vpc/module/variables.tf:524]: Now that I think about it more...
The consumer would normally modify the override.json to customize their environment. 
When used as a module, the override.json file will end up hidden as part of the module code, in `.terraform/modules/...` location. I'm not sure how a consumer would modify it at that point. 

Thoughts?

**Review by toddgiguere — CHANGES_REQUESTED** [review]

**Comment by toddgiguere** [comment]
After initial first-look review of individual elements, I also wanted to mention a couple more things:

When running an upgrade from main to this, for VPC pattern, the plan after upgrade reported:
```
Plan: 95 to add, 0 to change, 34 to destroy
```

I'm also unsure of what to do with the "override.json" file? In this solution it is put into the new submodule so that its loading from the config.tf still works. But I'm unsure if it belongs there. If a consumer needs to modify it when using the pattern directly, they might expect it at the `pattern/vpc` level, instead of a module. And if the consumer is using the module, the file will be buried in a hidden `.terraform/modules` directory and not very accessible.

**Comment by ocofaigh** [comment]
https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/pull/480 should be merged before this PR

**Comment by toddgiguere** [comment]
Overall I think this is the right approach, and the other two (roks and vsi) can be done similarly.

Holding off final review until the SCC v1 functionality is merged in.

**Comment by ocofaigh** [comment]
https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/pull/480 is merged - need to resolve conflicts in this PR. Also we should probably do roks and vsi here too?

**Review by toddgiguere — APPROVED** [review]

**Review by ocofaigh — CHANGES_REQUESTED** [review]
provider config needs to be removed from the new child modules (its in main.tf). It should only be in the pattern

**Review by ocofaigh — APPROVED** [review]

**Comment by surajsbharadwaj** [comment]
This file is now not required right ? https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/blob/vsi_module/patterns/vsi/module/catalogValidationValues.json.template

**Comment by ocofaigh** [comment]
@surajsbharadwaj That file does not exist in this PR. Looks like you are referencing an old branch (`vsi_module`)

**Pull request merged**
